src/builder.rs

```diff
@@ -43,15 +43,29 @@ pub enum Error {
     DuplicateSignature,
 }
 
-impl From<halo2_proofs::plonk::Error> for Error {
+/// An error type for adding a spend to the builder.
+#[derive(Debug, PartialEq, Eq)]
+pub enum SpendError {
```

Comment:
The new error types should implement `Display` to preserve their previous error messages. We should also implement `std::error::Error` for all of these error types (including `BuildError` which previously didn't have error messages). These changes would need to go into the changelog.

Reply:
Added!